raiden/blockchain_events_handler.py

```diff
@@ -171,6 +133,77 @@ def handle_channel_new(raiden: "RaidenService", event: Event):
     raiden.add_pending_greenlet(retry_connect)
 
 
+def create_light_channel(
+    block_hash,
+    block_number,
+    channel_identifier,
+    participant1,
+    participant2,
+    token_network_identifier,
+    transaction_hash,
+    raiden):
+    channel_state = create_channel(block_hash, block_number, channel_identifier, participant1, participant2, raiden,
+                                   token_network_identifier, transaction_hash)
+
+    partner_address = channel_state.partner_state.address
+
+    if ConnectionManager.BOOTSTRAP_ADDR != partner_address:
+        raiden.start_health_check_for(partner_address, participant1)
+
+
+def create_channel(block_hash, block_number, channel_identifier, participant1, participant2, raiden,
+                   token_network_identifier, transaction_hash):
+    channel_state = create_channel_state_and_proxy(block_number,
+                                                   channel_identifier,
+                                                   token_network_identifier,
+                                                   participant1,
+                                                   participant2,
+                                                   raiden)
+    new_channel = ContractReceiveChannelNew(
+        transaction_hash=transaction_hash,
+        channel_state=channel_state,
+        block_number=block_number,
+        block_hash=block_hash,
+    )
+    raiden.handle_and_track_state_change(new_channel)
+    return channel_state
+
+
+def create_channel_state_and_proxy(block_number,
+                                   channel_identifier,
+                                   token_network_identifier,
+                                   participant1: ChecksumAddress,
+                                   participant2: ChecksumAddress,
+                                   raiden):
+    is_participant1_handled_lc = LightClientService.is_handled_lc(to_checksum_address(encode_hex(participant1)),
+                                                                  raiden.wal)
+    is_participant2_handled_lc = LightClientService.is_handled_lc(to_checksum_address(encode_hex(participant2)),
+                                                                  raiden.wal)
+    is_light_channel = is_participant1_handled_lc or is_participant2_handled_lc
+    both_participants_are_light_clients = is_participant1_handled_lc and is_participant2_handled_lc
```

Comment:
Here we are checking for second time if particpant is lc. This could be improved

Reply:
you can use this `and` as an in-line assignment on line `202` instead

Reply:
the same applies to the variable set on `182`